src/psyclone/tests/parse/module_manager_test.py

```diff
@@ -0,0 +1,314 @@
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Author: J. Henrichs, Bureau of Meteorology
+
+
+'''Module containing py.test tests for the ModuleManager.'''
+
+import os
+
+import pytest
+
+from psyclone.errors import InternalError
+from psyclone.parse import ModuleManager
+
+
+@pytest.fixture(scope='function', autouse=True)
+def clear_module_manager_instance():
+    ''' The tests in this module all assume that there is no pre-existing
+    ModuleManager object, so this fixture ensures that the module manager
+    instance is deleted before and after each test function. The latter
+    makes sure that any other test executed next will automatically reload
+    the default ModuleManager file.
+    '''
+
+    # Enforce loading of the default ModuleManager
+    ModuleManager._instance = None
+
+    # Now execute all tests
+    yield
+
+    # Enforce loading of the default ModuleManager
+    ModuleManager._instance = None
+
+
+# ----------------------------------------------------------------------------
+def test_mod_manager_instance():
+    '''Tests the singleton functionality.'''
+    mod_man1 = ModuleManager.get()
+    mod_man2 = ModuleManager.get()
+    assert mod_man1 is mod_man2
+
+    with pytest.raises(InternalError) as err:
+        ModuleManager()
+
+    assert ("You need to use 'ModuleManager.get()' to get the singleton "
+            "instance." in str(err.value))
+
+
+# ----------------------------------------------------------------------------
+def test_mod_manager_get_modules_in_file():
+    '''Tests that file names are mapped as expected to the module they
+    contain. '''
+    mod_man = ModuleManager.get()
+    assert mod_man.get_modules_in_file("a_mod.f90") == ["a_mod"]
+    assert mod_man.get_modules_in_file("b_mod.F90") == ["b_mod"]
+    assert mod_man.get_modules_in_file("c_mod.x90") == ["c_mod"]
+    assert mod_man.get_modules_in_file("d_mod.X90") == ["d_mod"]
+    assert mod_man.get_modules_in_file("d.f90") == []
+
+
+# ----------------------------------------------------------------------------
+def mod_man_test_setup_directories():
```

Comment:
I mis-read that first, and thought adding it to `tests/conftest.py`, which I considered, but rejected because it's really only two tests who need this. But after re-reading, I realised that we can have `conftest.py` in subdirectories. That's neat, I forgot about that.